src/tools/genpolicy/src/registry.rs

```diff
@@ -252,130 +256,166 @@ async fn get_image_layers(
     let file_name = str::replace(&layer_digest, ":", "-");
-
     let mut decompressed_path = base_dir.join(file_name);
     decompressed_path.set_extension("tar");
 
     let mut compressed_path = decompressed_path.clone();
     compressed_path.set_extension("gz");
 
-    let mut verity_path = decompressed_path.clone();
-    verity_path.set_extension("verity");
-
     let mut verity_hash = "".to_string();
     let mut error_message = "".to_string();
     let mut error = false;
 
-    if use_cached_files && verity_path.exists() {
-        info!("Using cached file {:?}", &verity_path);
-    } else if let Err(e) = create_verity_hash_file(
-        use_cached_files,
-        client,
-        reference,
-        layer_digest,
-        &base_dir,
-        &decompressed_path,
-        &compressed_path,
-        &verity_path,
-    )
-    .await
-    {
-        error = true;
-        error_message = format!("Failed to create verity hash for {layer_digest}, error {e}");
+    // get value from store and return if it exists
+    if use_cached_files {
+        verity_hash = read_verity_from_store(&cache_file, &layer_digest)?;
+        info!("Using cache file");
+        info!("dm-verity root hash: {verity_hash}");
     }
 
-    if !error {
-        match std::fs::read_to_string(&verity_path) {
-            Err(e) => {
-                error = true;
-                error_message = format!("Failed to read {:?}, error {e}", &verity_path);
-            }
-            Ok(v) => {
-                verity_hash = v;
-                info!("dm-verity root hash: {verity_hash}");
+    // create the layer files
+    if !error && verity_hash == "" {
+        if let Err(e) = create_decompressed_layer_file(
+            client,
+            reference,
+            layer_digest,
+            &decompressed_path,
+            &compressed_path,
+        )
+        .await
+        {
+            error_message = format!(
+                "Failed to create verity hash for {layer_digest}, error {e}"
+            );
+            error = true
+        };
+
+        if !error {
+            match get_verity_hash_value(&decompressed_path) {
+                Err(e) => {
+                    error_message = format!("Failed to get verity hash {e}");
+                }
+                Ok(v) => {
+                    verity_hash = v;
+                    if use_cached_files {
+                        add_verity_to_store(&cache_file, &layer_digest, &verity_hash)?;
+                    }
+                    info!("dm-verity root hash: {verity_hash}");
+                }
             }
         }
     }
 
-    if !use_cached_files {
-        let _ = std::fs::remove_dir_all(&base_dir);
-    } else if error {
-        delete_files(&decompressed_path, &compressed_path, &verity_path);
-    }
-
+    temp_dir.close()?;
     if error {
-        panic!("{error_message}");
+        if Path::new(cache_file).exists() {
+            std::fs::remove_file(&cache_file)?;
```

Comment:
Let's think if we should delete the entire file, or just the verity hash for the current image layer.

Reply:
Whichever you prefer. I could see it going either way:
1. Delete the cache file for a fresh restart with (maybe?) a smaller chance of something messing up.
2. remove the single layer that errored out, so we don't lose cached work 

lmk what you think